File: java-example/src/main/java/no/javatec/pubsubemulator/kotest/spock/pubsub/DemoPublisher.java
package no.javatec.pubsubemulator.kotest.spock.pubsub;

import no.javatec.pubsubemulator.kotest.spock.dto.SampleReturnMessage;
import io.micronaut.gcp.pubsub.annotation.PubSubClient;
import io.micronaut.gcp.pubsub.annotation.Topic;

@PubSubClient
public interface DemoPublisher {

    @Topic("${pubsub.topic}")
    void send(SampleReturnMessage data);
}

File: kotlin-example/src/main/kotlin/no/javatec/pubsubemulator/kotest/pubsub/DemoPublisher.kt
package no.javatec.pubsubemulator.kotest.pubsub

import io.micronaut.gcp.pubsub.annotation.PubSubClient
import io.micronaut.gcp.pubsub.annotation.Topic
import no.javatec.pubsubemulator.kotest.dto.SampleReturnMessage

@PubSubClient
fun interface DemoPublisher {

    @Topic("\${pubsub.topic}")
    fun send(data: SampleReturnMessage)
}